Minecraft/src/main/java/com/microsoft/Malmo/Client/ClientStateMachine.java

```diff
@@ -546,6 +546,31 @@ else if (command.equals(killClientCommand))
                         if (config.getBoolean("replaceable", "runtype", false, "Will be replaced if killed"))
                         {
                             reply("MALMOOK", dos);
+
+                            missionPoller.stopServer();
+
+                            // Give non-hard exit 10 seconds to complete and force a hard exit.
+                            Thread deadMansHandle = new Thread(new Runnable() {
+                                @Override
+                                public void run() {
+                                    for (int i = 10; i > 0; i--) {
+                                        try {
+                                            Thread.sleep(1000);
+                                            System.out.println("Waiting to exit " + i + "...");
+                                        } catch (InterruptedException e) {
+                                            System.out.println("Interrupted " + i + "...");
+                                        }
+                                    }
+
+                                    // Kill it with fire!!!
+                                    System.out.println("Attempting hard exit");
```

Comment:
Is there any danger of getting here unnecessarily? eg if exitJava(0, true) works correctly, do we still get to run this code? (And would it matter if we did?)

Reply:
It should only run if the exitJava(, false) takes more than 10 seconds to run all the registered exit hooks. These run on other threads as part of the normal shutdown but I've seem the exit(, false) not return. Hopefully forcing the hard exit after 10 seconds will exit the process but could leave some files in a corrupted / incompletely written state so relying on next start or something else to clean up.